app/Http/Controllers/Frontend/FlightController.php

```diff
@@ -105,6 +106,22 @@ public function search(Request $request)
             Log::emergency($e);
         }
 
+        // Get only used Flight Types for the search form
+        // And filter according to settings
+        $usedtypes = DB::table('flights')->select('flight_type')->groupby('flight_type')->orderby('flight_type', 'asc');
```

Comment:
Use the `Flight` model for these, or even better, go through the repository so this gets cached. You can also follow the pattern I'm using in other places:

```php
$where = [];
if (setting('pilots.restrict_to_company')) {
  $where['airline_id'] = $user->airline_id;
}

if (setting(...)) {
  $where['dpt_airport_id'] = $user->curr_airport_id;
}

// etc
$used_types = Flight::where($where)->groupby(...);
```

Reply:
When i used the repository, it effected the results shown at the page too and gaved errors 'cause I was selecting only the flight_type ...

I am not sure what it will do if I use pluck('flight_type') though. This is why I used the DB instead of model or repository.

Reply:
Ah ok, I can look into the repository later. But it should still work with the `Flight` model. `pluck` from a collection will run the full query but it will then only do the `pluck` from the resulting collection, so the database won't be doing the filtering

Reply:
Works fine with the model, switching now (tested both options on local)